src/lib/CollisionPrevention/CollisionPreventionTest.cpp

```diff
@@ -0,0 +1,188 @@
+	{
+		uORB::Manager::initialize();
+		param_init();
+	}
+
+	void TearDown() override
+	{
+		param_reset_all();
+		uORB::Manager::terminate();
+	}
+};
+
+TEST_F(CollisionPreventionTest, testInstantiation) { CollisionPrevention cp(nullptr); }
+
+TEST_F(CollisionPreventionTest, testParamReadWrite)
+{
+	// GIVEN a parameter handle
+	param_t param = param_handle(px4::params::MPC_COL_PREV_D);
+
+	// WHEN: we get the parameter
+	float value = -999;
+	int status = param_get(param, &value);
+
+	// THEN it should be successful and have the default value
+	EXPECT_EQ(0, status);
+	EXPECT_EQ(-1, value);
+
+	// WHEN: we set the parameter
+	value = 42;
+	status = param_set(param, &value);
+
+	// THEN: it should be successful
+	EXPECT_EQ(0, status);
+
+	// WHEN: we get the parameter again
+	float value2 = -1999;
+	status = param_get(param, &value2);
+
+	// THEN: it should be the value we set
+	EXPECT_EQ(0, status);
+	EXPECT_EQ(42, value2);
+}
+
+TEST_F(CollisionPreventionTest, testUorbSendReceive)
+{
+	// GIVEN: a uOrb message
+	obstacle_distance_s message;
+	memset(&message, 0xDEAD, sizeof(message));
+	message.min_distance = 1.f;
+	message.max_distance = 10.f;
+
+	// AND: a subscriber
+	uORB::SubscriptionData<obstacle_distance_s> sub_obstacle_distance{ORB_ID(obstacle_distance)};
+
+	// WHEN we send the message
+	orb_advert_t obstacle_distance_pub = orb_advertise(ORB_ID(obstacle_distance), &message);
+	ASSERT_TRUE(obstacle_distance_pub != nullptr);
+
+	// THEN: the subscriber should receive the message
+	sub_obstacle_distance.update();
+	const obstacle_distance_s &obstacle_distance = sub_obstacle_distance.get();
+
+	// AND: the values we got should be the same
+	EXPECT_EQ(message.timestamp, obstacle_distance.timestamp);
+	EXPECT_EQ(message.min_distance, obstacle_distance.min_distance);
+	EXPECT_EQ(message.max_distance, obstacle_distance.max_distance);
+
+	// AND: all the bytes should be equal
+	EXPECT_EQ(0, memcmp(&message, &obstacle_distance, sizeof(message)));
+}
+
+TEST_F(CollisionPreventionTest, testBehaviorOff)
+{
+	// GIVEN: a simple setup condition
+	CollisionPrevention cp(nullptr);
+	matrix::Vector2f original_setpoint(10, 0);
+	float max_speed = 3;
+	matrix::Vector2f curr_pos(0, 0);
+	matrix::Vector2f curr_vel(2, 0);
+
+	// WHEN: we check if the setpoint should be modified
+	matrix::Vector2f modified_setpoint = original_setpoint;
+	cp.modifySetpoint(modified_setpoint, max_speed, curr_pos, curr_vel);
+
+	// THEN: it should be the same
+	EXPECT_EQ(original_setpoint, modified_setpoint);
+}
+
+TEST_F(CollisionPreventionTest, testBehaviorOn)
+{
+	// GIVEN: a simple setup condition
+	CollisionPrevention cp(nullptr);
+	matrix::Vector2f original_setpoint(10, 0);
+	float max_speed = 3;
+	matrix::Vector2f curr_pos(0, 0);
+	matrix::Vector2f curr_vel(2, 0);
+
+	// AND: a parameter handle
+	param_t param = param_handle(px4::params::MPC_COL_PREV_D);
+
+
+	// WHEN: we set the parameter check if the setpoint should be modified
+	float value = 10;
```

Comment:
It's a bit hard to read that test because I don't know what it means if this param is 10.